sqlserver/README.md

```diff
@@ -69,7 +69,16 @@ To configure this check for an Agent running on a host:
 
     See the [example check configuration][6] for a comprehensive description of all options, including how to use custom queries to create your own metrics.
 
-    **Note**: The (default) provider `SQLOLEDB` is being deprecated. To use the newer `MSOLEDBSQL` provider, set the `adoprovider` variable to `MSOLEDBSQL` in your `sqlserver.d/conf.yaml` file after having downloaded the new provider from [Microsoft][7]. It is also possible to use the Windows Authentication and not specify the username/password with:
+For hosts that are configured with the `<hostname\instance name>`, add `reported_hostname:` to the `sqlserver.d/conf.yaml` file, in the `conf.d/` at the root of the [Agent's configuration directory][5]:
```

Comment:
why do customers need to add `reported_hostname` here if they're setting `host` to be the same value?

Reply:
In the ticket that was being investigated, the customer was using `\` it was causing a connection error, when the `reported_hostname` was added it resolved the issue. Is this not longer true.

`datadog_checks.sqlserver.connection.SQLConnectionError: Unable to connect to SQL Server for instance lppsrtcprsql01,1433 - master: OperationalError('08001', '[08001] [Microsoft][ODBC SQL Server Driver][DBNETLIB]SQL Server does not exist or access denied. (17) (SQLDriverConnect); [08001] [Microsoft][ODBC SQL Server Driver][DBNETLIB]ConnectionOpen (Connect()). (53); [08001] [Microsoft][ODBC SQL Server Driver]Invalid connection string attribute (0)') `

Reply:
Yeah, I don't know why that would resolve the issue you saw? I think it's more likely something else changed bc `reported_hostname` is used only for reporting purposes & is not used in the connection string.  If that is true, we would want to update our main docs anyways https://docs.datadoghq.com/database_monitoring/setup_sql_server/

Can you link me the customer ticket so I can take a look?

Reply:
Here is the[ ticket](https://datadog.zendesk.com/agent/tickets/958937), There is also another ticket that we found the solution : [https://datadog.zendesk.com/agent/tickets/820353](https://datadog.zendesk.com/agent/tickets/820353)

Reply:
> Removing the port # and adding the 2nd line worked on both servers I did it to. It connects to the named instance now. I suggest that be updated in the documentation as an option

hmm I really doubt that adding `reported_hostname` changed anything about the connection. You can see [in the code](https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/connection.py#L579) we reference the `host` in the connection string, which comes from [here](https://github.com/DataDog/integrations-core/blob/d227b60e5eea49d39dc3b027c34f004eda20a840/sqlserver/datadog_checks/sqlserver/connection.py#L421) (they're running the 7.38 version). It seems more likely what worked about this was removing the port? 

It looks like in the 7.38 version of the agent, there was no code [like this](https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/connection.py#L449-L484), which defaults the port to be 1433 if not set. I'll have to double check if this will be a problem for them when they upgrade.

Reply:
Ah OK, so the correct thing to do is to set the port to 0 https://github.com/DataDog/integrations-core/pull/13135/files#diff-a38a2a09e86a943d0704a9f2922133905324820865ff052eff46209ee989d6d9R499

this works in 7.41 https://github.com/DataDog/integrations-core/blob/master/sqlserver/CHANGELOG.md#1001--2022-12-02--agent-7410 looks like agent versions 7.39-7.40 are broken for this use case. The recommended doc should be 

```yaml
instances:
 - host: <named>\<instance-name>
    port: 0
    ....
```

Reply:
https://github.com/DataDog/documentation/pull/16247